wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Local migration content provider

# Description

This PR implements a unified content provider for local data migration which will be useful in side-loading content as needed from the WordPress app to the Jetpack app.

### To test:

#### Prerequisites:

**Feature flags**: For convenience, user flags can be hard-coded by locally reverting this commit: https://github.com/wordpress-mobile/WordPress-Android/pull/17398/commits/c5773eeea2176433e8fb25951ff174096bb465f7

**Generate some local-only content**:

1. Install the WordPress app flavor from this branch
2. Login to the WordPress app and select a site
3. Publish a post
4. Make changes to the post and do not publish them (local changes)
5. Publish another post
6. Make changes on the web without syncing in the app
7. Make changes to the post in the app (conflict)

Log out of Jetpack and delete all data:

1. Force stop the app (if running)
2. Go to storage and clear app data

#### Testing steps:

1. Run the Jetpack app from this branch
2. Expect to be logged in automatically
3. Select the site where you have local-only content
4. Expect to see labels for local content in the post list

#### Automated tests

The automated tests added in this PR should also pass.

## Regression Notes
1. Potential unintended areas of impact
No. Existing implementations are changed, but are behind a feature flag.

9. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing implementations have automated tests, and manual testing was also done to ensure things are still working as desired.

10. What automated tests I added (or what prevented me from doing so)
Unit tests were added for these implementations.

PR submission checklist:
